### PR TITLE
j: 902-release-b -> 904-beta-c

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/j/default.nix [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
+  ];
 
-  doCheck = true;
-
-  # Causes build failure due to warning
-  hardeningDisable = lib.optional stdenv.cc.isClang "strictoverflow";
+  platform =
+    if (stdenv.isAarch32 || stdenv.isAarch64) then "raspberry"
+    else if stdenv.isLinux then "linux"
+    else if stdenv.isDarwin then "darwin"
```
Comment (7c6f434c) [inline pkgs/development/interpreters/j/default.nix:31]: If this piece of code is getting moved around anyway, maybe put the Darwin condition first now? Apparently there is now some M1 support

Comment (AndersonTorres) [reply pkgs/development/interpreters/j/default.nix:30]: Okay!

Comment (7c6f434c) [reply pkgs/development/interpreters/j/default.nix:30]: Isn't the raspberry case `isLinux`, though?

Comment (AndersonTorres) [reply pkgs/development/interpreters/j/default.nix:30]: If this is the case, then what? raspberry is irrelevant?
In any case Borg built both linux intel and arm 64 bits.

**Comment by AndersonTorres** [comment]
Now I will pass this to the Darwin mages.

File: pkgs/development/interpreters/j/default.nix (hunk 1 above)
Comment (7c6f434c) [reply pkgs/development/interpreters/j/default.nix:30]: If we have raspberry, it should be above Linux in the condition list (but maybe dropping it if aarch64 works anyway is also fine).

I would mostly keep the old logic and move Darwin up, so Darwin?ARM?Linux?

In the current expression keeping raspberry is pointless, I guess we can also try dropping it and see how it goes.

The current code just looks strange.

Comment (AndersonTorres) [reply pkgs/development/interpreters/j/default.nix:30]: Looking at the [original code](https://github.com/jsoftware/jsource/blob/4f422d11a4006c8c68db2ef815d871e88bc398e5/make2/build_all.sh#L21), it is a bit weirder. I think I will rewrite the Nix expression in order to match it.

Comment (AndersonTorres) [reply pkgs/development/interpreters/j/default.nix]: @7c6f434c now everything is working as expected!

Comment (7c6f434c) [reply pkgs/development/interpreters/j/default.nix]: I especially like the `j64x:=j32` case (upstream).

Thanks for your work!

**Pull request merged**